### PR TITLE
Fix Error When turtle or at random is preceded by space

### PR DESCRIPTION
**Pull request by racertop**
I edited `InvalidSpaceException` to add a `has_turtle` field, so when we repair this error we can decide to add the turtle code. Also import random module, so when we use `at random` command actually execute the code

**Fix for**

It fixes these 2 errors:
image
image

**How to test**

Just add a leading space to any code that includes turtle or at random, in any level:

` forward`

### CODE REVIEW

File: hedy.py [hunk 1]
```diff
@@ -1444,6 +1445,7 @@ def transpile_inner(input_string, level):
 
 def execute(input_string, level):
     python = transpile(input_string, level)
+    print(python)
```
Comment (Felienne) [inline hedy.py:1448]: I think you  might have added this for debugging? Can be removed in the final PR.

File: app.py [hunk 2]
```diff
@@ -336,7 +336,21 @@ def parse():
 
 def invalid_space_error_to_response(ex, translations):
     warning = translate_error(ex.error_code, translations, vars(ex))
-    code = "# coding=utf8\n" + ex.fixed_code
+    if ex.has_turtle:
```
Comment (Felienne) [inline app.py:339]: This is now duplicated with lines 290 and further, maybe we can store this is a constant called TURTLE_PREFIX_CODE or something?

**Comment by Felienne** [comment]
Lovely fixes @racertop! That will make it a lot clearer for kids. I added a few small suggestions to the code to make it a bit easier to process for others.

**Review by Felienne — CHANGES_REQUESTED** [review]
See above

File: hedy.py (hunk 1 above)
Comment (racertop) [reply hedy.py:1448]: Yes, sorry, this can be removd! I actually removed most of them and forgot thid one.

Comment (Felienne) [reply hedy.py:1448]: No problem, this has happened to me often too! Will you do it or do you want me to?

Comment (racertop) [reply hedy.py:1448]: Ill do it, Im having conection problems but Ill upload it.

File: app.py (hunk 2 above)
Comment (racertop) [reply app.py:339]: Understood I'm changing it and will upload the changes!

**Comment by racertop** [comment]
Added requested changes @Felienne! Hope it's ok now, thank you for your suggestions.

**Comment by racertop** [comment]
@Felienne can you not merge the test and the semantics :( I wasnt ready to commit those files, the editor added them. Those are for the other things Im working on, sorry for the problems, if you want me to uncommit the changes I can do that as well.

File: tests/tests_level_01.py [hunk 3]
```diff
@@ -330,8 +330,11 @@ def test_non_keyword_with_argument_gives_invalid(self):
       exception=hedy.InvalidCommandException,
       test_name=self.name()
     )
-
-
+  # This will only test valid programs
```
Comment (racertop) [inline tests/tests_level_01.py:333]: Error, dont merge this

File: SEMANTICS.md [hunk 4]
```diff
@@ -84,5 +84,80 @@ Level 1 supports:
   </tr>
  
 
+## Level 2
```
Comment (racertop) [inline SEMANTICS.md:87]: WIP, dont merge this either :cry: :sweat:

**Comment by rix0rrr** [comment]
No worries. To undo your changes to those files, type these commands one by one:

```
git checkout main tests/tests_level_01.py SEMANTICS.md
git commit -am 'Removing some files from the PR'
git push -u
```

If you are not using the git command line, then it depends on your client and I don't know 😅. In that case, look up the old contents of the files on GitHub and copy/paste them back.

Good luck!

**Comment by racertop** [comment]
Thanks for the help, @rix0rrr ! I copied and pasted from the GitHub main. If there's anything else I need to do, please tell me. It won't happen next time, I'm sorry.

File: app.py (hunk 2 above)
Comment (Felienne) [reply app.py:343]: Thanks!!

**Review by Felienne — APPROVED** [review]

**Pull request merged**